src/twisted/internet/defer.py

```diff
@@ -159,6 +165,65 @@ def maybeDeferred(f, *args, **kw):
         return succeed(result)
 
 
+def unbox(deferred, raises=True, passthrough=True):
+    """
+    Unbox a L{Deferred} that has already been fully processed.
```

Comment:
I wasn't completely happy with the "has already fired its result" because of the behavior of:

```python

d = Deferred()
d2 = Deferred()

d.addCallback(lambda _: d2)
d.callback(None)
```

Has the ``Deferred`` ``d`` "fired" its result? It's had ``callback()`` called on it so it seems to me like it had, but it's been chained to another ``Deferred`` that has not yet fired its result. Ultimately I don't really care much what the terminology is, I was just worried about that being confusing.